terragrunt/modules/docs-rs/builder.tf

```diff
@@ -0,0 +1,44 @@
+// The instance profile the builder will assume when communicating with
+// other AWS services.
+
+resource "aws_iam_instance_profile" "builder" {
+  name = "builder"
+  role = aws_iam_role.builder.name
+}
+
+resource "aws_iam_role" "builder" {
+  name = "builder"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Principal = {
+          Service = "ec2.amazonaws.com"
+        }
+        Action = ["sts:AssumeRole"]
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy" "builder_s3" {
+  role = aws_iam_role.builder.name
+  name = "builder_s3"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      // Access to s3
+      {
+        Effect = "Allow"
+        Action = "s3:*"
+
+        Resource = [
+          aws_s3_bucket.storage.arn,
+          "${aws_s3_bucket.storage.arn}/*"
```

Comment:
Is there a way to make this append-only, so it can't delete or read existing files?

Reply:
Yes, we can set the Action above to only non-delete write operations.

Reply:
I tried moving to only allowing the following actions:

PutObject, PutObjectTagging, PutObjectAcl, CreateBucket

This failed in a strange way. All artifacts were created and successfully uploaded to the s3. The logs only showed one error which simply stated that the library in question could not be documented (even though it clearly was successfully being done so). There were no other errors in the logs as far as I could see to indicate some additional error. Any idea what might be happening here?